src/utils/isStatelessComponent.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+import isReactModuleName from './isReactModuleName';
+import isReactComponentClass from './isReactComponentClass';
+import isReactCreateClassCall from './isReactCreateClassCall';
+import match from './match';
+import recast from 'recast';
+import resolveToModule from './resolveToModule';
+import resolveToValue from './resolveToValue';
+
+var {types: {namedTypes: types}} = recast;
+
+const validPossibleStatelessComponentTypes = [
+  'Property',
+  'FunctionDeclaration',
+  'FunctionExpression',
+  // TODO: maybe include these variants for safety:
+  // https://github.com/benjamn/ast-types/blob/master/def/es6.js#L31
+  // https://github.com/estree/estree/issues/2
+  // 'ArrowExpression', 'ArrowFunction'
+  'ArrowFunctionExpression'
+];
+
+function containsJSXElementOrReactCreateElementCall(node) {
+  var visited = false;
+  recast.visit(node, {
+    visitJSXElement(path) {
+      visited = true;
```

Comment:
Mmh, that would also consider `foo` in 

```javascript
function foo() {
  return function bar() {
    return <div />;
  }
}
```

as React component, i.e. any function that contains JSX is considered to be a React component. I wonder if we can narrow this down more.